primitives/src/circuit/merkle_tree/mod.rs

```diff
@@ -46,7 +70,7 @@ where
     /// Type to represent the merkle path of the concrete MT instantiation.
     /// It is MT-specific, e.g arity will affect the exact definition of the
     /// Merkle path.
-    type MerklePathVar;
+    type MerkleProofVar;
```

Comment:
> * the code comment is not updated (still use "merkle path")
> * can we have names that's more aligned with their non-circuit counterpart.
>   
>   * `type LeafVar` -> `type ElemVar`

I agree we should aim for consistency with native impl.
Maybe we should change the native impl naming though? I feel like Leaf is much more descriptive than Element.
A worry I have is that for example `create_leaf_variable` -> `create_element_variable` becomes pretty confusing, at least to me.

Reply:
interesting proposal, so you are saying we introduce `Leaf` concept in native impl? let's discuss this over a call today @mrain @tessico.

Reply:
> okay I notice a mismatch of concepts.
> 
> in mt impl, `verify(&self, pos, proof)` the `elem` is assumed to be part of `proof`; whereas our mt_circuit impl, `is_member(leaf, proof, root)` where `leaf = uid + elem` and `proof` doesn't have `elem`?
> 
> we should make them consistent
> 
> * i think `is_member(root_var, uid_var, proof_var)` may seem longer, but clearer and less error prone IMHO. and matches with non-circuit impl, where `&self` corresponds to `root_var`.
> * making `MerkleProofVar` inclusive of `elem_var` makes it more consistent

Addressed these in https://github.com/EspressoSystems/jellyfish/pull/172/commits/59e321c6b87dab4e65c647afd995eea344e76310. Good suggestions, thx @alxiong!

For now I've taken the liberty to also rename the fields in circuit impl from `elem` -> `leaf`. Since the usage was already mixed (leaf & elem) inside the circuit impl, at least here we can have consistency. TBD is whether the native impl should switch to `leaf` (or otherwise leaf here back to `elem`), but I think this can be a separate change, separate PR?

Reply:
I generally like your change (esp removing `LeafVar`)
but my confusion is: 

```rust
pub struct Merkle3AryMembershipProofVar {
    node_vars: Vec<Merkle3AryNodeVar>, // <-- this should be called path_var instead?
    leaf_var: Variable, // <-- why is this called leaf_var instead of elem_var
}
```

and since we already removed concept of "leaf" for user-facing interfaces, (I'm fine with internal variable named `leaf_xxx` to refer to the union of `uid` and `elem`) , why we still use `leaf_idx_var` instead of `elem_idx_var`?